src/dc_receive_imf.rs

```diff
@@ -1793,7 +1802,28 @@ async fn create_or_lookup_mailinglist(
         ),
     };
 
+    if let Some(list_post) = &mime_parser.list_post {
+        let (contact_id, _) =
+            Contact::add_or_lookup(context, "", list_post, Origin::Hidden).await?;
+        let mut contact = Contact::load_from_db(context, contact_id).await?;
+        contact.param.set(Param::ListPost, &listid);
+        contact.update_param(context).await?;
+    }
+
     if let Some((chat_id, _, blocked)) = chat::get_chat_id_by_grpid(context, &listid).await? {
+        if let Some(list_post) = &mime_parser.list_post {
+            let mut chat = Chat::load_from_db(context, chat_id).await?;
+
+            if let Some(old_list_post) = chat.param.get(Param::ListPost) {
+                if list_post != old_list_post {
+                    // Apparently the mailing list is using a different List-Post header in each message.
+                    // Make the mailing list read-only because we would't know which message the user wants to reply to.
+                    chat.param.remove(Param::ListPost);
```

Comment:
Maybe set it to empty string instead to indicate that list has broken `List-Post`?

This way the `if let Some(old_list_post) = chat.param.get(Param::ListPost)` line can have an `else` clause that adds `ListPost` parameter to existing mailing lists created in previous delta chat versions.

Reply:
I implemented this, but when testing I noticed that it doesn't work because `lookup_chat_by_reply()` assigns a chat id before and this function is not executed.

As this will also lead to other problems, I'll need to create a `apply_mailinglist_changes` function.

Reply:
Fixed by 6c05189, regression test in 658998a.